onnxscript/optimizer/_constant_folding.py

```diff
@@ -1104,23 +1116,31 @@ def fold_constants(
     onnx_shape_inference: bool = False,
     input_size_limit: int = DEFAULT_CONSTANT_FOLD_INPUT_SIZE_LIMIT,
     output_size_limit: int = DEFAULT_CONSTANT_FOLD_OUTPUT_SIZE_LIMIT,
-) -> bool:
+) -> FoldConstantsResult:
     """
     Applies constant folding optimization to the model.
-    Returns true iff the model was modified.
+
+    Args:
+        model (ir.Model): The ONNX model to optimize.
+        external_data_folder (str, optional): Path to the folder containing external data
+            for the model. Defaults to an empty string.
+        onnx_shape_inference (bool, optional): Whether to enable ONNX shape inference during
+            constant folding. Defaults to False.
+        input_size_limit (int, optional): The maximum size (in bytes) of input tensors
```

Comment:
```suggestion
        model : The ONNX model to optimize.
        external_data_folder: Path to the folder containing external data
            for the model. Defaults to an empty string.
        onnx_shape_inference: Whether to enable ONNX shape inference during
            constant folding. Defaults to False.
        input_size_limit: The maximum size (in bytes) of input tensors
```

No need to add types in docstring as we have type annotations already